### PR TITLE
Allow for persisting plugin configuration changes.

### PR DESCRIPTION
**Pull request by UserNotFound**
Using `RABBITMQ_ENABLED_PLUGINS_FILE`, we can tell RabbitMQ where to find this file, and store it on the persistent volume.

https://www.rabbitmq.com/relocate.html

You can view available plugins with the command `rabbitmq-plugins list`

To enable (persistently) an available plugin, from within the database container run `rabbitmq-plugins enable $PLUGIN --online`. This will turn on the plugin, and it will persist after a restart.

---
This change is [Reviewable](https://reviewable.io/reviews/aptible/docker-rabbitmq/30)

### CODE REVIEW

**Comment by UserNotFound** [comment]
@joshraker Candidate for internal documentation!

File: Dockerfile.erb [hunk 1]
```diff
@@ -7,9 +7,9 @@ ENV         TAG=<%= ENV.fetch 'TAG' %> \
             RABBITMQ_LOGS=- \
             RABBITMQ_SASL_LOGS=-
 ENV         RABBITMQ_HOME=/srv/rabbitmq_server-${RABBITMQ_VERSION} \
-            PLUGINS_DIR=/srv/rabbitmq_server-${RABBITMQ_VERSION}/plugins \
-            ENABLED_PLUGINS_FILE=/srv/rabbitmq_server-${RABBITMQ_VERSION}/etc/rabbitmq/enabled_plugins
```
Comment (almathew) [inline Dockerfile.erb]: @UserNotFound what's the reason for the env var name change here? 
I vaguely remember talking about this at some point, although I don't completely remember the reasoning.

Comment (almathew) [reply Dockerfile.erb]: Specifically, I know that both of those vars have value to RabbitMQ, I'm just not sure why we're preferring one over the other here.

Comment (UserNotFound) [reply Dockerfile.erb]: I'm not sure why I didn't have a comment here but this was marked as resolved.  Specifically, 
`ENABLED_PLUGINS_FILE` has no value to Rabbit (the proper ENV is `RABBITMQ_ENABLED_PLUGINS_FILE` per their docs) and we weren't referencing our variable anywhere ourselves.

**Comment by joshraker** [comment]
Is there a reason we're using `grep '\[E\*\]'` all over the place as opposed to `fgrep` or `grep -F '[E*]'`? I feel like the latter makes it easier to understand what's going on.

**Comment by UserNotFound** [comment]
> Is there a reason we're using `grep '\[E\*\]'` all over the place as opposed to `fgrep` or `grep -F '[E*]'`? I feel like the latter makes it easier to understand what's going on.

Good point! Pushing grep -F change.

File: bin/run-database.sh [hunk 2]
```diff
@@ -108,6 +108,10 @@ bootstrap_configuration () {
 
 if [[ "$1" == "--initialize" ]]; then
     /usr/bin/initialize-certs
+
+    # Start with the default plugins, but persist them on the data volume
+    cp /tmp/enabled_plugins_template "${DATA_DIRECTORY}/enabled_plugins"
```
Comment (joshraker) [inline bin/run-database.sh]: `${DATA_DIRECTORY}/enabled_plugins` == `${RABBITMQ_ENABLED_PLUGINS_FILE}`

```suggestion
    cp /tmp/enabled_plugins_template "${RABBITMQ_ENABLED_PLUGINS_FILE}"
```

Comment (UserNotFound) [reply bin/run-database.sh]: This was intentional, to avoid defining the location twice. Is there a downside to doing it this way I might have overlooked?

Comment (UserNotFound) [reply bin/run-database.sh]: Thanks!

File: Dockerfile.erb [hunk 3]
```diff
@@ -7,9 +7,9 @@ ENV         TAG=<%= ENV.fetch 'TAG' %> \
...
-ENV         PATH=$RABBITMQ_HOME/sbin:$PATH
+            PLUGINS_DIR=/srv/rabbitmq_server-${RABBITMQ_VERSION}/plugins
+ENV         RABBITMQ_ENABLED_PLUGINS_FILE=${DATA_DIRECTORY}/enabled_plugins \
```
[the remark this answers is not in the document]
Comment (UserNotFound) [reply Dockerfile.erb]: You're right - I believe I meant to leave this `ENV` declaration on the next line, for PATH.

File: test-plugins.sh [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+echo "Starting DB"
+docker run -it --rm -d --name="$DB_CONTAINER" \
+  --volumes-from "$DATA_CONTAINER" -h aptible \
+  "${IMG}" &> /dev/null
+
+echo "Waiting for DB"
+wait_for_rabbitmq
+
+echo "Testing for existing plugin"
+docker exec -it "$DB_CONTAINER" rabbitmq-plugins list rabbitmq_management | grep '\[E\*\]' &> /dev/null
```
Comment (joshraker) [inline test-plugins.sh]: You caught the `grep -F` changes in `test/rabbitmq.bats` but not in this file.

**Pull request merged**